quantum/dip_switch.c

```diff
@@ -70,16 +84,22 @@ __attribute__((weak)) bool dip_switch_update_mask_kb(uint32_t state) {
 
 void dip_switch_init(void) {
 #ifdef DIP_SWITCH_PINS
-#    if defined(SPLIT_KEYBOARD) && defined(DIP_SWITCH_PINS_RIGHT)
+#   ifdef SPLIT_KEYBOARD
     if (!isLeftHand) {
-        const pin_t dip_switch_pad_right[] = DIP_SWITCH_PINS_RIGHT;
         for (uint8_t i = 0; i < NUMBER_OF_DIP_SWITCHES; i++) {
             dip_switch_pad[i] = dip_switch_pad_right[i];
+#           if defined(SPLIT_KEYBOARD) && defined(DIP_SWITCH_INVERT_RIGHT)
+                dip_switch_invert[i] = dip_switch_invert_right[i];
+#           endif
         }
     }
-#    endif
+#   endif
     for (uint8_t i = 0; i < NUMBER_OF_DIP_SWITCHES; i++) {
-        setPinInputHigh(dip_switch_pad[i]);
+        if (dip_switch_invert[i]) {
+            setPinInputLow(dip_switch_pad[i]);
```

Comment:
This kind of code won't compile for AVR, because those MCUs don't support internal pull-downs on their GPIO pins, therefore `setPinInputLow()` fails at compile time. You would need to wrap it in some preprocessor conditionals, so that `setPinInputLow()` is not used unless `DIP_SWITCH_INVERT` is actually defined.

Reply:
I changed this to be wrapped in the preprocessor conditional